src/components/PrivacyModal/PrivacyModal.tsx

```diff
@@ -0,0 +1,151 @@
+  const navigate = useNavigate()
+  const [open, setOpen] = useState(true)
+  const [checked, setChecked] = useState(false)
+  const { privacyPolicyCookie, handleAccept } = usePrivacyModal()
+  const currentLocation = useLocation()
+
+  //Disable backdropClick so the Modal only closes via the buttons
+  const handleClose = useCallback((_: React.MouseEvent<HTMLElement>, reason: string) => {
+    if (reason && reason == 'backdropClick') return
+  }, [])
+
+  //setting the check so the button gets enabled
+  const handleChecked = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      setChecked((event.target as HTMLInputElement).checked)
+    },
+    [setChecked]
+  )
+
+  //close the modal and set the cookie in the hook
+  const handleModal = useCallback(
+    (cookieAccepted: boolean) => {
+      handleAccept(cookieAccepted)
+      setOpen(false)
+    },
+    [handleAccept, setOpen]
+  )
+
+  //load nothing if cookie is set or privacypolicy page is loaded
+  if (privacyPolicyCookie || currentLocation.pathname.includes('privacypolicy')) return null
+
+  return (
+    <Modal
+      aria-labelledby="privacy-modal"
+      aria-describedby="makes-the-user-accept-the-privacypolicy-or-not"
+      open={open}
+      onClose={handleClose}
+      closeAfterTransition>
+      <Box sx={style}>
+        <Typography id="transition-modal-title" variant="h4" component="h2">
+          {t('components.PrivacyModal.termsOfService')}
+          <Typography id="modal-text" variant="subtitle1" component="h2">
+            {t('components.PrivacyModal.afterReading')}
+            <FormGroup>
+              <FormControlLabel
+                onClick={handleChecked}
+                control={<Checkbox />}
+                label={
+                  <>
+                    {t('components.PrivacyModal.readPrivacypolicy')}&nbsp;
+                    <Link
+                      marginX="0.2em"
+                      component="button"
+                      variant="subtitle1"
+                      color={'textSecondary'}
+                      href={'/privacypolicy'}
+                      underline="hover"
+                      onClick={() => navigate('/privacypolicy')}>
+                      {t('pages.PrivacyPolicy')}
+                    </Link>
+                    {t('components.PrivacyModal.readPrivacypolicy2')}
+                  </>
+                }
+              />
+            </FormGroup>
+          </Typography>
+          <Box
+            height={'20%'}
+            sx={{
+              display: 'flex',
+              justifyContent: 'space-between'
+            }}>
+            <Tooltip title={t('components.PrivacyModal.returnToMoodle')}>
+              <Button
+                variant={'contained'}
+                sx={{ alignSelf: 'end' }}
+                aria-multiline={'true'}
+                onClick={() => {
+                  handleModal(false), (window.location.href = 'https://moodle.hs-kempten.de/enrol/index.php?id=357')
```

Comment:
Ich hab' mal nachgeschaut und finde, dass .assign() ein bisschen besser zum Wechseln einer Webseite passt. Bei href wird die aktuelle Webseite überschrieben. Das wirkt etwas hacky.

Habe mit Dimitri gesprochen, da es irgendwie nicht so sinnvoll ist, alle Studierenden, also auch Informatiker, auf das Game Engineering SWEG Moodle zu schicken. Also auch die Aschaffenburger, wenn sie unser Frontend verwenden. Wir würden vorschlagen, die Hochschule aus dem Backend zu fetchen und dann mittels if else die User zu verschiedenen Moodles zu schicken. (Zwischen IF und GE soll nach der Evaluation aber irgendwann dennoch entschieden werden, um Studierende direkt dorthin zu schicken, wo sie herkamen) 

Die Keys der Hochschulen sind HS-KE und TH-AB
Die Links sind https://moodle.hs-kempten.de/ und https://moodle.th-ab.de/


```suggestion
                  handleModal(false)
                  window.location.assign(https://moodle.hs-kempten.de/enrol/index.php?id=357')
```

Reply:
Ich hab gesehen, du hast das wohl schon mal zwischen href und assign hin und her geändert. Der Linter meckert bei href auch rum, da hier eine ungewollte Mutation durchgeführt wird.